### PR TITLE
Switch to JWTs

### PR DESCRIPTION
**Pull request by arthurian**
This PR switches the auth method to JWTs instead of opaque tokens. The changes should be transparent to library users if they have been using the `Client` class as described in the `README`. 

Notes:
- Removed the `obtain_token` internal API methods.
- Added `create_jwt ` method to generate JWTs. 
- Modified `client.authenticate()` to generate a JWT for the user. If a `course_id` is specified, then it will hit the `authorize-user` endpoint on behalf of the user.
- Updated tests.
- Added requirement that PyJWT be at least 1.7.1 (preferably >2.0.0). The minimum version is based on the version currently used by the _hxat_ application.

### CODE REVIEW

File: media_management_sdk/api.py [hunk 1]
```diff
@@ -43,10 +43,14 @@ def headers(self) -> Dict[str, str]:
...
             "Content-Type": "application/json",
         }
-        if self.access_token is not None:
-            headers["Authorization"] = f"Token {self.access_token}"
+        if self.authorization_header:
+            headers["Authorization"] = self.authorization_header
         return headers
 
+    @property
+    def authorization_header(self) -> str:
```
Comment (Tanvez) [inline media_management_sdk/api.py:51]: cool function annotation! Never seen this before.

**Review by Tanvez — APPROVED** [review]
LGTM!

File: media_management_sdk/jwt.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            course_permission if course_permission in ("read", "write") else "read"
+        )
+
+    token = jwt.encode(payload, client_secret, algorithm="HS256")
+
+    # back-compat: clients using PyJWT <2.0.0 forces bytes, so try to decode as unicode
+    try:
+        token = token.decode() # type: ignore
+    except (UnicodeDecodeError, AttributeError):
+        pass
```
Comment (dodget) [inline media_management_sdk/jwt.py:50]: excellent

**Review by dodget — APPROVED** [review]
Looks good

**Pull request merged**